src/Data/Singletons/Promote/Defun.hs

```diff
@@ -177,18 +192,11 @@ tyFunArity (DArrowT `DAppT` (DConT tyFunNm `DAppT` _ `DAppT` b) `DAppT` DStarT)
   = 1 + tyFunArity b
 tyFunArity _ = 0
 
--- Checks if type is (TyFun a b -> *)
-isTyFun :: DKind -> Bool
-isTyFun (DArrowT `DAppT` (DConT tyFunNm `DAppT` _ `DAppT` _) `DAppT` DStarT)
-  | tyFunName == tyFunNm
-  = True
-isTyFun _ = False
```

Comment:
Amusingly, there appears to be an `isTyFun` function that may have been used at one point to distinguish the `(->)` in `TyFun a b -> Type` from other uses of `(->)`. But at the moment, it's not used anywhere in `singletons`, and with this patch, it becomes unnecessary.